src/my-bot.ts

```diff
@@ -1,38 +1,423 @@
+                    Column: i
+                });
+            }
+        } else {
+            for (let i: number = shipPlace.StartingSquare.Row.charCodeAt(0); i <= shipPlace.EndingSquare.Row.charCodeAt(0); i++) {
+                ship.push({
+                    Row: String.fromCharCode(i),
+                    Column: shipPlace.StartingSquare.Column
+                });
+            }
+        }
+        return ship;
+    }
+
+    private generateAdjacentSquares(ship: Position[]): Position[] {
+        if (ship === []) {
+            return []
+        }
+        let squares: Position[] = [];
+        for (let i: number = 0; i < ship.length; i++) {
+            squares.push({ Row: ship[i].Row, Column: ship[i].Column + 1 });
+            squares.push({ Row: ship[i].Row, Column: ship[i].Column - 1 });
+            squares.push({ Row: String.fromCharCode(ship[i].Row.charCodeAt(0) + 1), Column: ship[i].Column });
+            squares.push({ Row: String.fromCharCode(ship[i].Row.charCodeAt(0) - 1), Column: ship[i].Column });
+            squares.push({ Row: String.fromCharCode(ship[i].Row.charCodeAt(0) + 1), Column: ship[i].Column + 1 });
+            squares.push({ Row: String.fromCharCode(ship[i].Row.charCodeAt(0) - 1), Column: ship[i].Column + 1 });
+            squares.push({ Row: String.fromCharCode(ship[i].Row.charCodeAt(0) + 1), Column: ship[i].Column - 1 });
+            squares.push({ Row: String.fromCharCode(ship[i].Row.charCodeAt(0) - 1), Column: ship[i].Column - 1 });
+        }
+        return squares;
+    }
+
+    private randomShot(gamestate, hits: Position[] = []): Position {
+        const useless: Position[] = this.generateAdjacentSquares(hits);
+        let position: Position = { Row: 'A', Column: 1 };
+        let found: boolean = false;
+        do {
+            let row: string = ""
+            let column: number = 0;
+            if (Math.random() < 0.5) {
+                row = String.fromCharCode(Math.floor(Math.random() * 5) * 2 + 65);
+                column = Math.floor(Math.random() * 5) * 2 + 1;
+            } else {
+                row = String.fromCharCode(Math.floor(Math.random() * 5) * 2 + 66);
+                column = Math.floor(Math.random() * 5) * 2 + 2;
+            }
+            position = { Row: row, Column: column};
+            found = false;
+            for (let i = 0; i < gamestate.MyShots.length; i++) {
+                if ((gamestate.MyShots[i].Position.Row === position.Row) && (gamestate.MyShots[i].Position.Column === position.Column)) {
+                    found = true;
+                    break;
+                }
+            }
+            for (let i = 0; i < useless.length; i++) {
+                if ((useless[i].Row === position.Row) && (useless[i].Column === position.Column)) {
+                    found = true;
+                    break;
+                }
+            }
+        } while (found)
+        return position
+    }
+
+    private track(gamestate, snapCopy): Position {

```

Comment:
This is another really long method. I think you could almost:

* Make this a class called `TrackState` or similar
* Set up your variables as private vars
* Cut it into a bunch of methods (`setup`, `getDirection`, etc.)

...and all of these would be a lot easier to follow.